lib/puppet/type/iis_application_pool.rb

```diff
@@ -1,48 +1,395 @@
     end
   end
-  
-  newproperty(:managedpipelinemode, :parent => PuppetX::PuppetLabs::IIS::Property::String) do
+
+  newproperty(:auto_start, :boolean => true) do
+    newvalue(:true)
+    newvalue(:false)
+       
+    munge do |value|
+      resource.munge_boolean(value)
+    end
+  end
+
+  newproperty(:clr_config_file, :parent => PuppetX::PuppetLabs::IIS::Property::String) do
+  end
+
+  newproperty(:enable32_bit_app_on_win64, :boolean => true) do
+    newvalue(:true)
+    newvalue(:false)
+       
+    munge do |value|
+      resource.munge_boolean(value)
+    end
+  end
+
+  newproperty(:enable_configuration_override, :boolean => true) do
+    newvalue(:true)
+    newvalue(:false)
+       
+    munge do |value|
+      resource.munge_boolean(value)
+    end
+  end
+
+  newproperty(:managed_pipeline_mode, :parent => PuppetX::PuppetLabs::IIS::Property::String) do
     desc "The managedPipelineMode of the ApplicationPool. First letter has to be capitalized."
     validate do |value|
-      unless value.kind_of?(String)
-        fail("Invalid value '#{value}'. Should be a string")
-      end
+      super value
       unless ['Integrated','Classic'].include?(value)
         fail("Invalid value '#{value}'. Valid values are Integrated or Classic")
       end
     end
   end
-  
-  newproperty(:managedruntimeversion, :parent => PuppetX::PuppetLabs::IIS::Property::String) do
+
+  newproperty(:managed_runtime_loader, :parent => PuppetX::PuppetLabs::IIS::Property::String) do
+  end
+
+  newproperty(:managed_runtime_version, :parent => PuppetX::PuppetLabs::IIS::Property::String) do
     desc "The managedRuntimeVersion of the ApplicationPool."
     validate do |value|
-      unless value.kind_of?(String)
-        fail("Invalid value '#{value}'. Should be a string")
-      end
+      super value
       unless ['v4.0','v2.0'].include?(value)
-        fail("Invalid value '#{value}'. Valid values are Integrated or Classic")
+        fail("Invalid value '#{value}'. Valid values are v4.0 or v2.0")
+      end
+    end
+  end
+
+  newproperty(:pass_anonymous_token, :boolean => true) do
+    newvalue(:true)
+    newvalue(:false)
+       
+    munge do |value|
+      resource.munge_boolean(value)
+    end
+  end
+
+  newproperty(:start_mode, :parent => PuppetX::PuppetLabs::IIS::Property::String) do
+    validate do |value|
+      super value
+      unless ['OnDemand','AlwaysRunning'].include?(value)
+        fail("Invalid value '#{value}'. Valid values are OnDemand or AlwaysRunning")
+      end
+    end
+  end
+
+  newproperty(:queue_length, :parent => PuppetX::PuppetLabs::IIS::Property::PositiveInteger) do
+    validate do |value|
+      super value
+      fail "#{self.name.to_s} should be greater than 10" unless value.to_i > 10
+      fail "#{self.name.to_s} should be less than or equal to 65535" unless value.to_i <= 65535
+    end
+  end
+
+  newproperty(:cpu_action, :parent => PuppetX::PuppetLabs::IIS::Property::String) do
+    validate do |value|
```

Comment:
You can use `newvalue` here instead of custom code. See [docs](https://docs.puppet.com/guides/custom_types.html#specifying-allowed-values) and [coding examples](https://github.com/puppetlabs/puppetlabs-apt/blob/master/lib/puppet/type/apt_key.rb#L52). This will expose the information to the docs processor.

How stable are those values in MS' API, and how often will we have to release updates to keep up? Does the IIS API provide good error messages we could re-use instead of validating ourselves?

From a UX point of view: would it be helpful to be case-insensitive on the puppet side here?